inference-engine/thirdparty/clDNN/src/gpu/loop_gpu.cpp

```diff
@@ -0,0 +1,399 @@
+/*
+// Copyright (c) 2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+*/
+
+#include "loop_inst.h"
+#include "network_impl.h"
+#include "implementation_map.h"
+#include "math_utils.h"
+#include "register_gpu.hpp"
+#include "mutable_data_inst.h"
+#include "input_layout_inst.h"
+#include "memory_impl.h"
+#include <vector>
+#include <algorithm>
+
+namespace cldnn {
+namespace gpu {
+struct loop_gpu : typed_primitive_impl<loop> {
+    const loop_node& node;
+
+    explicit loop_gpu(const loop_node& node) : node(node) {}
+
+    static memory_impl::ptr get_external_memory(const loop_inst& instance, const primitive_id& external_id) {
+            const auto outputPrim = instance.get_network().get_primitive(external_id);
```

Comment:
nit: alignment

Reply:
`get_external_memory()` is moved to `loop_inst` and the alignment issue is fixed.